Auto/Tactic.lean

```diff
@@ -284,13 +284,18 @@ def queryTPTP (exportFacts : Array REntry) : LamReif.ReifM (Array Embedding.Lam.
 
 open Embedding.Lam in
 def querySMT (exportFacts : Array REntry) (exportInds : Array MutualIndInfo) : LamReif.ReifM (Option Expr) := do
-  let lamVarTy := (← LamReif.getVarVal).map Prod.snd
+  -- GEORGE: do we need to pass more of `LamReif:State` to `lamFOL2SMT`?
+  let lamVarTy ← LamReif.getVarVal
+  trace[auto.lamReif.printValuation] "lamVarTy: {lamVarTy}"
   let lamEVarTy ← LamReif.getLamEVarTy
+  trace[auto.lamReif.printValuation] "lamEVarTy: {lamEVarTy}"
+  let tyVal ← LamReif.getTyVal
+  trace[auto.lamReif.printValuation] "tyVal: {tyVal}"
   let exportLamTerms ← exportFacts.mapM (fun re => do
     match re with
     | .valid [] t => return t
     | _ => throwError "runAuto :: Unexpected error")
-  let commands ← (lamFOL2SMT lamVarTy lamEVarTy exportLamTerms exportInds).run'
+  let commands ← (lamFOL2SMT lamVarTy lamEVarTy tyVal exportLamTerms exportInds).run'
```

Comment:
We're threading a lot of the `LamReif` `State` into `lamFOL2SMT` and associated functions. Would it be cleaner to make this part of relevant monad?

If yes, could you suggest how to do that? I haven't worked with monads in Lean before.

Reply:
Your approach looks good to me. Using a monad might make things cleaner, but I'm not sure. Anyways, I'll clean up the code when you're done, so you can keep going for now.